smpp/transmitter.go

```diff
@@ -255,6 +259,14 @@ func (t *Transmitter) Submit(sm *ShortMessage) (*ShortMessage, error) {
 	f.Set(pdufield.ScheduleDeliveryTime, sm.ScheduleDeliveryTime)
 	f.Set(pdufield.ReplaceIfPresentFlag, sm.ReplaceIfPresentFlag)
 	f.Set(pdufield.SMDefaultMsgID, sm.SMDefaultMsgID)
+
+	//set the optional parameters in the submit pdu from sm
+	optParams := p.TLVFields()
+
+	for param, value := range sm.OptsParams {
+		optParams[param] = value
+	}
+
```

Comment:
Let's follow the standard of `go-smpp` and use `optParams.Set()` just like above with Fields.